lib/connect.js

```diff
@@ -101,6 +101,11 @@ function connect(url, socketOptions, openCallback) {
   var parts = URL.parse(url, true); // yes, parse the query string
   var protocol = parts.protocol;
   var noDelay = !!sockopts.noDelay;
+  // We should default to true as this nicely complements the heartbeat
+  var keepAlive = !!sockopts.keepAlive || true;
```

Comment:
This doesn't actually do what you want... If I specify ```{ keepAlive: false }```, this will basically say ``` var keepAlive = false || true;```, which obviously is always true. So you're basically always enabling keepAlive here.